src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/PlayerDeathSystem.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.ligthandshadow.componentsystem.controllers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.EventPriority;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.ligthandshadow.componentsystem.LASUtils;
+import org.terasology.ligthandshadow.componentsystem.components.LASTeamComponent;
+import org.terasology.logic.characters.AliveCharacterComponent;
+import org.terasology.logic.characters.CharacterComponent;
+import org.terasology.logic.characters.CharacterTeleportEvent;
+import org.terasology.logic.health.BeforeDestroyEvent;
+import org.terasology.logic.health.DoHealEvent;
+import org.terasology.logic.players.PlayerCharacterComponent;
+
+
+@RegisterSystem
+public class PlayerDeathSystem extends BaseComponentSystem {
+    Logger logger = LoggerFactory.getLogger(PlayerDeathSystem.class);
+
+    @ReceiveEvent(priority = EventPriority.PRIORITY_HIGH)
+    public void beforeDestroy(BeforeDestroyEvent event, EntityRef player, CharacterComponent characterComponent, AliveCharacterComponent aliveCharacterComponent) {
+        if (player.hasComponent(PlayerCharacterComponent.class)) {
+            event.consume();
+            logger.info(player.toFullDescription());
+            String team = player.getComponent(LASTeamComponent.class).team;
+            player.send(new DoHealEvent(100000, player));
```

Comment:
@darshan3 what does the magic number 100000 mean? it would be better to use a constant.

Reply:
I took it from healthMax command. https://github.com/MovingBlocks/Terasology/blob/develop/modules/Core/src/main/java/org/terasology/logic/health/HealthCommands.java

Reply:
@darshan3 find out the value which restores the health completely and use that as a local constant for now.
Might ask @e-aakash to set that up as a constant somewhere in the health module and reference it later.

Reply:
The Health module should have a way to `restoreToMax` - not sure how/whether a constant can solve this.

Reply:
Sure, extracting the DoHeal event logic with the maxHealth value to such a method would be ideal.

Reply:
If we make a constant in health component, we would anyway need to access the component to get the value. We can either use `maxHealth` for now or just this _hack_ of arbitrarily large value. 
`restoreFullEvent` is part of the present plan for the Restoration system. So, an event to restore to max health would be available when Restoration is complete (around the end of the second month of the coding period).